docs/src/implemented-proposals/multi-level-multi-iterator-scheduler.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Multi-Level Multi-Iterator Scheduler
+---
+
+## Background
+
+We have proposed graph-based scheduling in the past, however, as a hold-over until we implement those the current scheduler has been modified to use a multi-iterator approach. This approach significantly improves single-threaded throughput in situations with high-contention - and lends itself to a straight-forward stepping stone to graph-based scheduling, with a few modifications.
+
+## Single-Level Multi-Iterator
+
+The single-level multi-iterator has already been implemented in Solana's independent banking stage thread schedulers, this section serves to outline that concept in order to provide context for the multi-level multi-iterator.
+
+First, we take our priority-queue and collect the entire buffer into a vector. We initialize multiple, in this case 64, positions such that there are no write-conflicts between the transactions at the given positions. These, up to 64, transactions then make up a transaction batch to be executed. The multi-iterator marks these transactions as `already_processed`, and the positions are marched forward, again checking for conflicts as we set them.
```

Comment:
> The multi-iterator marks these transactions as `already_processed`

This is only marked *after* the transactions are successfully processed?

Reply:
Used the wrong name, we changed it to `already_handled` in the scanner - it's marked as `already_handled` if it gets rejected (non-retryable errors like sanitization, too many locks, etc), or if it's selected for execution.